src/js/media-controller.js

```diff
@@ -1140,7 +1140,19 @@ const getStreamType = (controller) => {
   
   if (!media) return undefined;
   
-  if (media.streamType) return media.streamType;
+  if (media.streamType) {
+    // If the slotted media supports `streamType` but
+    // `streamType` is "unknown", prefer `default-stream-type`
+    // if set (CJP)
+    if (media.streamType === 'unknown' && controller.hasAttribute('default-stream-type')) {
+      const defaultType = controller.getAttribute('default-stream-type');
+
+      if (StreamTypeValues.includes(defaultType)) {
+        return defaultType;
+      }
+    }
+    return media.streamType;
+  }
```

Comment:
just wondering why not gate it here and still try to derive it from the duration?

```suggestion
  if (media.streamType && media.streamType !== 'unknown') {
    return media.streamType;
  }
```

Reply:
@luwes I think that's conflating two different things. The derived case is intended for slotted media elements that do not support the `streamType` api. `default-stream-type` is a media chrome concept representing "what we want to treat as the stream type when it is not known". If someone has implemented `streamType` in their extended media element, I think we should treat that as the "source of truth". So, in this case, the slotted media definitely has an implementation of `streamType` (and therefore we should not override their implementation in media chrome), but the stream type is currently unknown for their implementation (and therefore we should use `default-stream-type` if set).